### PR TITLE
Introduce Flux Script Wizard

### PR DESCRIPTION
**Pull request by chnn**
Removes the `BodyBuilder`, and introduces the `FluxScriptWizard`.

flux script wizard

### CODE REVIEW

File: ui/src/reusable_ui/components/dropdowns/Dropdown.tsx [hunk 1]
```diff
@@ -123,8 +123,17 @@ class Dropdown extends Component<Props, State> {
...
 
     const selectedChild = children.find(child => child.props.id === selectedID)
-    const dropdownLabel =
-      (selectedChild && selectedChild.props.children) || titleText
+    const isLoading = status === ComponentStatus.Loading
+
+    let dropdownLabel
+
+    if (isLoading) {
+      dropdownLabel = <div className="dropdown--loading" />
```
Comment (ischolten) [inline ui/src/reusable_ui/components/dropdowns/Dropdown.tsx]: nice!

File: ui/src/shared/components/TimeMachine/FluxQueryMaker.tsx [hunk 2]
```diff
@@ -65,116 +31,91 @@ interface Props {
...
-            source={source}
-            queries={queries}
-            timeRange={timeRange}
-          />
+          >
+            {draftScript.trim() === '' && (
+              <div className="flux-script-wizard--bg-hint">
+                <p>
+                  New to Flux? Give the{' '}
+                  <a title="Open Script Wizard" onClick={this.handleShowWizard}>
```
Comment (ischolten) [inline ui/src/shared/components/TimeMachine/FluxQueryMaker.tsx]: NICE!

File: ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+
+// Types
+import {RemoteDataState, Source} from 'src/types'
+
+// This constant is selected so that dropdown menus will not overflow out of
+// the `.flux-script-wizard--wizard` window
+const DROPDOWN_MENU_HEIGHT = 110
+
+// A map from a database name to rention policies names, e.g.
+//
```
Comment (ischolten) [inline ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx]: i dont think you need to leave this comment in

File: ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+  measurements: string[]
+  measurementsStatus: RemoteDataState
+  selectedMeasurement: string | null
+  fields: string[]
+  fieldsStatus: RemoteDataState
+  selectedFields: string[] | null
+}
+
+export default class FluxScriptWizard extends PureComponent<Props, State> {
+  public state: State = {
```
Comment (ischolten) [inline ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx]: is there a reason for not having a constructor here?

File: ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+
+    this.setState({
+      fields,
+      fieldsStatus: RemoteDataState.Done,
+      selectedFields: [],
+    })
+  }
+}
+
+async function fetchDBsToRPs(proxyLink: string): Promise<DBsToRPs> {
```
Comment (ischolten) [inline ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx]: how do you feel about moving the functions outside the component to a utils file?

File: ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
+  fields.sort()
+
+  return fields
+}
+
+function formatDBwithRP(db: string, rp: string): string {
+  return `${db}/${rp}`
+}
+
+function toComponentStatus(state: RemoteDataState): ComponentStatus {
```
Comment (ischolten) [inline ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx]: this is great!

File: ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+  if (selectedFields.length) {
+    const fieldsPredicate = selectedFields
+      .map(f => `r._field == "${f}"`)
+      .join(' or ')
+
+    filterPredicate += ` and (${fieldsPredicate})`
+  }
+
+  const from = `from(bucket: "${selectedBucket}")`
+  const range = `|> range(start: -1h)`
```
Comment (ischolten) [inline ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx]: should this use `dashboardTime`?

File: ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx (hunk 4 above)
Comment (chnn) [reply ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx]: ¯\_(ツ)_/¯ it works the same way

File: ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx (hunk 7 above)
Comment (chnn) [reply ui/src/shared/components/TimeMachine/FluxScriptWizard.tsx]: Russ said to leave it as 1 hour for now

File: ui/src/shared/components/TimeMachine/FluxQueryMaker.tsx [hunk 8]
```diff
@@ -200,126 +148,49 @@ class FluxQueryMaker extends PureComponent<Props, State> {
...
-  private updateBody = async () => {
+  private checkDraftScript = async () => {
     const {draftScript} = this.state
 
-    let ast: object
+    if (draftScript.trim() === '') {
+      // Don't attempt to validate an empty script
+      this.setState({draftScriptStatus: VALID_SCRIPT_STATUS})
+
+      return
```
Comment (AlirieGray) [inline ui/src/shared/components/TimeMachine/FluxQueryMaker.tsx]: do we need this return?

**Pull request merged**